app/models/location.rb

```diff
@@ -56,4 +63,27 @@ def self.find_or_create_by_api_ids(locationiq_id, osm_id, osm_type)
       create_from_params(resp)
     end
   end
+
+  # Restrict Human location specificity to State, Country. Return new Location if restriction added.
+  def self.check_and_restrict_specificity(location, host_genome_name)
+    # We don't want Human locations with subdivision or city
+    if host_genome_name == "Human" && (location.subdivision_name.present? || location.city_name.present?)
+      # Redo the search for just the country/state
+      success, resp = geosearch_by_country_and_state(location.country_name, location.state_name)
+      unless success
+        raise "Couldn't find #{location.state_name}, #{location.country_name} (state, country)"
+      end
+
+      result = LocationHelper.adapt_location_iq_response(resp[0])
```

Comment:
Cant an empty `[]` response be a success?

Reply:
good catch 👍